.buildkite/common.py

```diff
@@ -14,6 +16,94 @@
 
 DEFAULT_KERNELS = ["linux_4.14", "linux_5.10"]
 
+# Define frequently used keywords having double quotes to avoid
+# having manual errors while updating them at different places
+# Mandotory: specify the instances on which to run the step
+INSTANCE="instances"
+# Mandotory: specify the kernel version on which to run the step
+KERNEL="kernels"
+# one of command/wait should be present in the steps
+COMMAND="commands"
+WAIT="wait"
+# optional if you want to add some cool emoji to the label
+EMOJI="emoji"
+# optional use in steps where artifacts needs to be uploaded
+ARTIFACTS="artifact_paths"
+# optional if a timeout is needed
+TIMEOUT="timeout"
+# optional if you want to identify the step with a specific label
+LABEL="label"
+
+def prepare_buildkite_step(test_step, label, instance, kv):
+    """
+    Parse test_step and create a step that
+    buildkite can understand
+
+    """
+    # get a copy to edit and cleanup the data
+    step_copy = copy.deepcopy(test_step)
+
+    # add the instance and kernel version to the step
+    step_copy["agents"] = [f"instance={instance}", f"kv={kv}"]
+
+    if LABEL in step_copy.keys():
+        step_copy[LABEL] = step_copy[LABEL] + label
+    else:
+        step_copy[LABEL] = label
+
+    # adding "key" is needed to search ARTIFACTS that
+    # are uploaded by other steps
+    step_copy["key"] = step_copy[LABEL]
+
+    if EMOJI in step_copy.keys():
+        step_copy[LABEL] = step_copy[EMOJI] + step_copy[LABEL]
+        # cleanup keys that are invalid as per buildkite step
+        del step_copy[EMOJI]
+
+    for i, _ in enumerate(step_copy[COMMAND]):
+        # this is so that artifacts can be download from a unique step
+        if "--step" in step_copy[COMMAND][i]:
+            step_copy[COMMAND][i] = step_copy[COMMAND][i].format(label)
```

Comment:
This and `key` above are just working around what could be a simple fix in the tests

Reply:
added some explanation here https://github.com/firecracker-microvm/firecracker/pull/3458#discussion_r1114530928